flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar/ArraySortFunction.java

```diff
@@ -0,0 +1,141 @@
+
+import static org.apache.flink.table.api.Expressions.$;
+
+/** Implementation of ARRAY_SORT function. */
+@Internal
+public class ArraySortFunction extends BuiltInScalarFunction {
+
+    private final ArrayData.ElementGetter elementGetter;
+    private final SpecializedFunction.ExpressionEvaluator greaterEvaluator;
+
+    private transient MethodHandle greaterHandle;
+
+    public ArraySortFunction(SpecializedFunction.SpecializedContext context) {
+        super(BuiltInFunctionDefinitions.ARRAY_SORT, context);
+        final DataType elementDataType =
+                ((CollectionDataType) context.getCallContext().getArgumentDataTypes().get(0))
+                        .getElementDataType();
+        elementGetter = ArrayData.createElementGetter(elementDataType.getLogicalType());
+        greaterEvaluator =
+                context.createEvaluator(
+                        $("element1").isGreater($("element2")),
+                        DataTypes.BOOLEAN(),
+                        DataTypes.FIELD("element1", elementDataType.notNull().toInternal()),
+                        DataTypes.FIELD("element2", elementDataType.notNull().toInternal()));
+    }
+
+    @Override
+    public void open(FunctionContext context) throws Exception {
+        greaterHandle = greaterEvaluator.open(context);
+    }
+
+    public @Nullable ArrayData eval(ArrayData array, Boolean... ascendingOrder) {
+        try {
+            if (array == null || ascendingOrder.length > 0 && ascendingOrder[0] == null) {
+                return null;
+            }
+            if (array.size() == 0) {
+                return array;
+            }
+            boolean isAscending = ascendingOrder.length > 0 ? ascendingOrder[0] : true;
+            Object[] elements = new Object[array.size()];
+            for (int i = 0; i < array.size(); i++) {
+                elements[i] = elementGetter.getElementOrNull(array, i);
+            }
+            if (isAscending) {
+                Comparator<Object> ascendingComparator = new MyComparator(true);
+                Arrays.sort(elements, ascendingComparator);
+            } else {
+                Comparator<Object> ascendingComparator = new MyComparator(false);
+                Arrays.sort(elements, ascendingComparator);
+            }
+            return new GenericArrayData(elements);
+        } catch (Throwable t) {
+            throw new FlinkRuntimeException(t);
+        }
+    }
+
+    private class MyComparator implements Comparator<Object> {
+        private final boolean isAscending;
+
+        public MyComparator(boolean isAscending) {
+            this.isAscending = isAscending;
+        }
+
+        @Override
+        public int compare(Object o1, Object o2) {
+            try {
+                if (isAscending) {
+                    if (o1 == null) {
+                        return -1;
+                    }
+                    if (o2 == null) {
+                        return -1;
+                    }
```

Comment:
Let's use `Comparator.nullsLast` and `Comparator.nullsFirst` instead.

Reply:
Ok